incubator/mariadb-cluster/templates/secrets.yaml

```diff
@@ -1,28 +1,38 @@
+{{- if (not .Values.root.existingSecret) -}}
 apiVersion: v1
 kind: Secret
 metadata:
   name: {{ template "fullname" . }}
   labels:
     app: "{{ template "name" . }}"
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    chart: {{ template "mariadb.chart" . }}
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
   {{- if .Values.root.password }}
   mariadb-root-password: "{{ .Values.root.password | b64enc }}"
-  {{- else }}
+  {{- else if (not .Values.root.forcePassword) }}
   mariadb-root-password: "{{ randAlphaNum 10 | b64enc }}"
+  {{ else }}
+  mariadb-root-password: {{ required "A MariaDB Root Password is required!" .Values.root.password }}
   {{- end }}
   {{- if .Values.db.user }}
   {{- if .Values.db.password }}
   mariadb-password: "{{ .Values.db.password | b64enc }}"
-  {{- else }}
+  {{- else if (not .Values.db.forcePassword) }}
   mariadb-password: "{{ randAlphaNum 10 | b64enc }}"
+  {{- else }}
+  mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.db.password }}
   {{- end }}
   {{- end }}
+  {{- if .Values.replication.enabled }}
```

Comment:
I believe we should remove this `if` statement since we always want to configure the master to support replication

Reply:
well, it would be enabled by default but I think the user want to have the option to not use replication and a single master, e.g. for testing